service/service.go

```diff
@@ -897,24 +928,26 @@ func (t *service) updateRecordsFromLog(tid thread.ID, lid peer.ID) {
 }
 
 // createLog creates a new log with the given peer as host.
-func createLog(host peer.ID) (info thread.LogInfo, err error) {
-	sk, pk, err := crypto.GenerateEd25519Key(rand.Reader)
-	if err != nil {
-		return
+func createLog(host peer.ID, key crypto.Key) (info thread.LogInfo, err error) {
+	var ok bool
+	if key == nil {
+		info.PrivKey, info.PubKey, err = crypto.GenerateEd25519Key(rand.Reader)
```

Comment:
Similar to thread follow key, a log private key is created on the service if _no_ log key was given (private / public)

Reply:
Ok, gotcha...